OrbitClientData/ModuleData.cpp

```diff
@@ -0,0 +1,100 @@
+// Copyright (c) 2020 The Orbit Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+#include "OrbitClientData/ModuleData.h"
+
+#include "OrbitBase/Logging.h"
+#include "OrbitClientData/FunctionUtils.h"
+#include "absl/synchronization/mutex.h"
+#include "capture_data.pb.h"
+
+using orbit_client_protos::FunctionInfo;
+
+FunctionInfo* ModuleData::FindFunctionByRelativeAddress(uint64_t relative_address,
+                                                        bool is_exact) const {
+  absl::MutexLock lock(&functions_mutex);
+  if (functions_.empty()) return nullptr;
+
+  uint64_t address = relative_address + load_bias();
+
+  if (is_exact) {
+    auto it = functions_.find(address);
+    return (it != functions_.end()) ? it->second.get() : nullptr;
+  }
+
+  auto it = functions_.upper_bound(address);
+  if (it == functions_.begin()) return nullptr;
+
+  --it;
+  FunctionInfo* function = it->second.get();
+  CHECK(function->address() <= address);
+
+  if (function->address() + function->size() < address) return nullptr;
+
+  return function;
+}
+
+void ModuleData::LoadSymbols(const orbit_grpc_protos::ModuleSymbols& module_symbols,
+                             uint64_t module_start_address) {
+  CHECK(!is_loaded());
```

Comment:
move it after the lock? you modify it there after all. Also maybe check the flag since the function will need to lock too.

Reply:
that makes sense, done